cdap-app-templates/cdap-etl/cdap-etl-batch/src/test/java/co/cask/cdap/template/etl/batch/ETLStreamConversionTest.java

```diff
@@ -77,6 +86,49 @@ public void testStreamConversionTPFSAvroSink() throws Exception {
     testSink(sinkType);
   }
 
+  @Test
+  public void testAvroSourceConversionToAvroSink() throws Exception {
+    String filesetName = "converted_stream";
+    StreamManager streamManager = getStreamManager("myStream");
+    streamManager.createStream();
+    streamManager.send(ImmutableMap.of("header1", "bar"), "AAPL|10|500.32");
+
+    ETLBatchConfig etlConfig = constructETLBatchConfig(filesetName, "TPFSAvro");
+    AdapterConfig adapterConfig = new AdapterConfig("description", TEMPLATE_ID.getId(), GSON.toJsonTree(etlConfig));
+    Id.Adapter adapterId = Id.Adapter.from(NAMESPACE, "sconversion");
+    AdapterManager manager = createAdapter(adapterId, adapterConfig);
+
+    manager.start();
+    manager.waitForOneRunToFinish(4, TimeUnit.MINUTES);
+    manager.stop();
+
+    // get the output fileset, and read the parquet/avro files it output.
+    DataSetManager<TimePartitionedFileSet> fileSetManager = getDataset(filesetName);
+    TimePartitionedFileSet fileSet = fileSetManager.get();
+
+    List<GenericRecord> records = readOutput(fileSet, EVENT_SCHEMA_1);
```

Comment:
looks like the first part of this test is only used to create an avro file. This is unnecessary and adds a couple minutes to the test. You can just create the input fileset in the test case directly, then write the input you want to write so that there is just a single mapreduce run in the test.